src/efftester.ml

```diff
@@ -185,11 +208,47 @@ let term_to_ocaml ?(typeannot = true) term =
       else Printf.bprintf sb "%s" x
     in
     match t with
+    | Constructor (_, name, payload_lst) ->
+      (match payload_lst with
+      | [] -> Printf.bprintf sb "%s" name
+      | trms ->
+        Printf.bprintf
+          sb
+          "(%s (%a))"
+          name
+          (* TODO: when we add support for payload that is a list with more than
+            one element, we need to fix the printing for payload list
+           *)
+            (fun sb lst -> List.iter (fun trm -> Printf.bprintf sb "%a" exp trm) lst)
```

Comment:
I would prefer to have the correct code right away; the only thing that needs to be changed is to have a comma-separated list of expressions.

I would recommend implementing a generic function:

```ocaml
val bprint_list :
  sep:(Buffer.t -> unit ->  unit) -> (Buffer.t -> 'a -> unit) -> Buffer.t -> 'a list -> unit
```

The `sep` argument is called on the buffer in-between each element of the input list (only when we know that an element follows), and the second argument is called on each list element. Then you can call it here as follows:

```ocaml
let sep buf () = Printf.fprintf buf ", " in
bprint_list ~sep exp sb lst
```

(Eventually we should convert the printer to use `Format`; I may send a PR myself after this one gets merged.)